File: 프로그래머스/unrated/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    var answer = 0;
    let unique = new Set(tangerine);
    let count = new Array();
    let countObj = {};
    tangerine.forEach((e) => {
        if (countObj[e] === undefined) {
            countObj[e] = 1;
        } else {
            countObj[e]++;
        }
    })
    
    unique.forEach((e) =>{
        count.push(countObj[e]);
    })
    count.sort((a,b) => b - a);
    let i = 0;
    while (k > 0) {
        k -= count[i];
        i++;
        answer++;
    }
    return answer;
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
from collections import Counter

def solution(k, tangerine):
    answer = 0
    count = list(Counter(tangerine).values())
    count.sort(reverse=True)
    i = 0
    while k > 0 and i < len(count):
        k -= count[i]
        i += 1
        answer += 1
    return answer